### PR TITLE
Unify sold2 config dataclasses

### LINKED ISSUE #2093: Update configs to use `dataclasses`
We have some specific configs for some algorithms, will be nice to update them from using `dicts`/`TypedDict` to `dataclasses`. 

The idea here is to do it in a way that does not break things, so we should have an interface (to/from) between `dict` and the `dataclasses`.

Example of what we can explore for these methods
```python
>>> from dataclasses import dataclass, asdict
>>> @dataclass
... class A:
...     b: int
... 
>>> asdict(A(1))
{'b': 1}
>>> A(**asdict(A(1)))
A(b=1)
```
_Originally posted by @johnnv1 in https://github.com/kornia/kornia/pull/2092#discussion_r1049564760_

List of some configs to be replaced:
 - [ ] kornia.feature.adalam.core.AdalamConfig
 - [ ] kornia.contrib.face_detection.FaceDetector.config #2851
 - [ ] kornia.feature.keynet.KeyNet_conf - #2254
 - [ ] [kornia.feature.loftr.loftr.default_cfg](https://github.com/kornia/kornia/blob/2387a2d165a977f1646332d6cbe6b915a4806e37/kornia/feature/loftr/loftr.py#L25)
   - [ ] kornia.feature.loftr.loftr_module.fine_preprocess.FinePreprocess.config
   - [ ] kornia.feature.loftr.loftr_module.transformer.LocalFeatureTransformer.config
   - [ ] kornia.feature.loftr.utils.coarse_matching.CoarseMatching.config
   - [ ] kornia.feature.loftr.utils.supervision config
   - [ ] [kornia.feature.loftr.backbone.resnet_fpn config](https://github.com/kornia/kornia/blob/2387a2d165a977f1646332d6cbe6b915a4806e37/kornia/feature/loftr/backbone/resnet_fpn.py#L50)
 - [ ] kornia.feature.matching._get_default_fginn_params
 - [ ] [kornia.feature.sold2.backbones.SOLD2Net.cfg](https://github.com/kornia/kornia/blob/2387a2d165a977f1646332d6cbe6b915a4806e37/kornia/feature/sold2/backbones.py#L377) - #2880
   - [ ] [kornia.feature.sold2.sold2.default_cfg](https://github.com/kornia/kornia/blob/2387a2d165a977f1646332d6cbe6b915a4806e37/kornia/feature/sold2/sold2.py#L18)
   - [ ] [kornia.feature.sold2.sold2_detector.default_cfg](https://github.com/kornia/kornia/blob/2387a2d165a977f1646332d6cbe6b915a4806e37/kornia/feature/sold2/sold2_detector.py#L17)

### PR DESCRIPTION
**Pull request by lappemic**
#### Changes
This PR refactors the newly introduced dataclasses in PR #2880 to a single file in `kornia/utils/structures.py`.

Fixes #2093 (according to [comment](https://github.com/kornia/kornia/pull/2880#pullrequestreview-2004735726))

ASAP this PR is closed i will refactor `kornia/feature/sold2` default_cfg to `structures.py` as well.

### CODE REVIEW

File: kornia/utils/structures.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+from dataclasses import dataclass, field
```
Comment (johnnv1) [inline kornia/utils/structures.py:1]: since it's a specific structure of sold2, rename it to `kornia/feature/sold2/structures.py`

Comment (lappemic) [reply kornia/utils/structures.py:1]: Oh, sorry, just saw that i missunderstood your initial comment! I move it!

**Comment by lappemic** [comment]
@johnnv1 applied the changes

**Review by johnnv1 — APPROVED** [review]

**Comment by johnnv1** [comment]
The doctest for kornia/feature/sold2/sold2_detector.py::kornia.feature.sold2.sold2_detector.SOLD2_detector is failling, could you check it? @lappemic

**Comment by lappemic** [comment]
@johnnv1 the docs are ok. I don't understand the other errors (e.g. Tests on CPU (PR) / tests-cpu (MacOS-latest, float32) ...) and also was not able to mitigate them. Do you know what the issue is there?

**Comment by johnnv1** [comment]
> @johnnv1 the docs are ok. I don't understand the other errors (e.g. Tests on CPU (PR) / tests-cpu (MacOS-latest, float32) ...) and also was not able to mitigate them. Do you know what the issue is there?

tks, these falling cases is some issue with the CI/torch

**Review by johnnv1 — APPROVED** [review]

**Comment by lappemic** [comment]
How can i mitigate them? Or how can i finish this PR then?

**Comment by johnnv1** [comment]
> How can i mitigate them? Or how can i finish this PR then?
We ping the guys -- @edgarriba @ducha-aiki could you force this merge? The red lights are unrelated within the PR

**Pull request merged**